src/doc/unstable-book/src/library-features/asm.md

```diff
@@ -743,11 +744,7 @@ Some registers cannot be used for input or output operands:
 | Hexagon | `lr` | This is the link register which cannot be used as an input or output. |
 | AVR | `r0`, `r1`, `r1r0` | Due to an issue in LLVM, the `r0` and `r1` registers cannot be used as inputs or outputs.  If modified, they must be restored to their original values before the end of the block. |
 
-In some cases LLVM will allocate a "reserved register" for `reg` operands even though this register cannot be explicitly specified. Assembly code making use of reserved registers should be careful since `reg` operands may alias with those registers. Reserved registers are the frame pointer and base pointer
-- The frame pointer and LLVM base pointer on all architectures.
-- `r9` on ARM.
-- `x18` on AArch64.
-- `r0` and `r1` on AVR.
+In some cases LLVM will allocate a "reserved register" for `reg` operands even though this register cannot be explicitly specified. Assembly code making use of reserved registers should be careful since `reg` operands may alias with those registers. Reserved registers that can sometimes be allocated are the frame pointer and base pointer in the list above.
```

Comment:
This is a little light on the explanation.
* How can I, as a user, "be careful" against register aliasing?
* If a reserved register is allocated into my registers how can I test for that? How can I fix it from happening again?

Reply:
I suppose the wording here is bad. What I meant to say is that `reg` may sometimes allocate the frame/base pointer register, even though it is not possible to allocate this register explicitly with `"rbp"`.

Reply:
That answers one question, but I still don't understand the following:
* If I type a register class, such as `reg`, can that give me a register LLVM considers reserved.
* If so, is that a problem, or will LLVM sort it out somehow?
* If LLVM doesn't hand that on its own, what is the mitigation that an end programmer can put in their code?

Reply:
So what is actually happening here is that in LLVM the frame pointer and base pointer registers are only marked as reserved if the function requires a frame/base pointer. If a register is not reserved then it may be selected by the register allocator.

In rustc we can't know in advance whether a function will require a frame/base pointer so we always disallow specifying the frame/base pointer as an explicit register. However it could still be selected for `reg` if the function doesn't need a frame/base pointer.